_posts/2016-10-13-contributing.md

```diff
@@ -0,0 +1,82 @@
+
+I'm a huge fan of the economics of open source, and I'm always pushing the work we do in the wider open source community, this got me thinking... 
+
+How, as a consultancy do we visualise our own digital footprint? Why don't we track, measure and visualise this, as we do so well for our customers? we thought we would have a little go at creating something exciting, the aim being we could put something that represents our digital footprint and the things we do in the wider community. 
+
+# What to measure
+
+How could we visualise such a thing? we wanted to know:
+
+* *How do we interact with open source code?* We want to see key metrics for our primary public open-source platform, Github. we're particularly interested in the volume of open-source public code we host on github, interactions such as forks/stars of these repositories.
+* *How are people using the stuff we contribute?* We want to see key metrics for our most popular distribution mechanisms, currently that's primarily the Docker Hub.
+* *How many people are viewing our website and reading out engineering blogs?* We don't need a wealth of metrics here but a few simple values would help is track and encourage engagement.
+
+Let's start with these simple metrics to ascertain our current levels of community interaction. Ultimately we want to use this data to track and encourage more open source collaboration across Infinity Works.
+
+## Strong Foundations
+
+Personally, I've spent a lot of time working with metrics recently, we decided upon the following tools to help me visualise this data:
+
+* [Prometheus](https://prometheus.io/) - Used as my scraping engine with a solid time series database to boot
+* [Grafana](http://grafana.org/) - Who else does graphs better?
+
+## Mapping in the data
+
+We're off to a good start, but it's time to get our hands dirty to get at the information we need. Neither Github nor Docker Hub have native Prometheus support, there also wasn't an option we could find within the community.
+We set out to create some tools for us and others to use, should they find them useful... (yay opensource!) 
+
+* Github Exporter ([Docker Hub](https://hub.docker.com/r/infinityworks/github-exporter/), [GitHub](https://github.com/infinityworksltd/github-exporter))
+* Docker Hub Exporter ([Docker Hub](https://hub.docker.com/r/infinityworks/docker-hub-exporter/), [GitHub](https://github.com/infinityworksltd/docker-hub-exporter))
+
+Whilst simple, the above two exporters helps us capture the metrics from the APIs and present them for scraping by the prometheus server. We can use these metrics to measure our contribution back to the community.
+
+Github fortunately has a simple, well documented API to retrieve the information you need. We made us of the Prometheus client libraries and some python to extract the data we needed from repositories and organisations.
```

Comment:
Fortunately, Github has a simple, well documented API to retrieve the information we needed. We made use of the Prometheus client libraries and some python to extract the data we needed from repositories and organisations.